### PR TITLE
Caching can now also handle complete directories

### PR DESCRIPTION
**Pull request by lievendoclo**
Optimized caching

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,7 +75,8 @@
     "ts-node": "^8.0.3",
     "tslint": "^5.14.0",
     "typedoc": "^0.14.2",
-    "typescript": "^3.4.1"
+    "typescript": "^3.4.1",
+    "uuid": "^3.3.2"
```
Comment (cdupuis) [inline package.json]: Why are these here? You are not using uuid. There’s a guid() function that should be used instead.

Comment (lievendoclo) [reply package.json]: I am, in the tests. But i'll replace it.
